fairing/preprocessors/full_notebook.py

```diff
@@ -7,16 +7,21 @@
 class FullNotebookPreProcessor(BasePreProcessor):
     def __init__(self,
                  notebook_file=None,
-                 command=["jupyter", "nbconvert", "--stdout", "--to", "notebook", "--execute"],
+                 input_files=None,
+                 command=["jupyter", "nbconvert", "--stdout", "--to", "notebook", "--execute", "--allow-errors",
+                          "--ExecutePreprocessor.timeout=-1"],
                  path_prefix=constants.DEFAULT_DEST_PREFIX,
                  output_map=None):
-        
+
         if notebook_file is None and notebook_util.is_in_notebook():
             notebook_file = notebook_util.get_notebook_name()
-        
+
+        input_files = input_files or []
+        input_files.append(notebook_file)
```

Comment:
can you check if the notebook_file is already in input_files and then add it? more general question, does input_files should be a set instead of a list? I understand that users don't use set regularly but internally we should be using set instead of list.

Reply:
Added the check to make sure notebook_file is already in input_files.

The base preprocessor handles conversion of the list of input files to a set, and also normalizes relative paths to avoid duplicates. When the list of input files is passed to the BasePreProcessor constructor, the duplicates will be removed.